src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/client/BitbucketCloudClient.java

```diff
@@ -0,0 +1,110 @@
+        String body = getObjectMapper().writeValueAsString(request);
+        Request req = new Request.Builder()
+                .put(RequestBody.create(APPLICATION_JSON_MEDIA_TYPE, body))
+                .url(format("%s/2.0/repositories/%s/%s/commit/%s/reports/%s", baseUrl(), project, repository, commit, REPORT_KEY))
+                .build();
+
+        LOGGER.info("Create report on bitbucket cloud");
+        LOGGER.debug("Create report: " + body);
+
+        try (Response response = getClient().newCall(req).execute()) {
+            validate(response);
+        }
+    }
+
+    void createAnnotations(String project, String repository, String commit, Set<CloudAnnotation> annotations) throws IOException {
+        if (annotations.isEmpty()) {
+            return;
+        }
+
+        Request req = new Request.Builder()
+                .post(RequestBody.create(APPLICATION_JSON_MEDIA_TYPE, getObjectMapper().writeValueAsString(annotations)))
+                .url(format("%s/2.0/repositories/%s/%s/commit/%s/reports/%s/annotations", baseUrl(), project, repository, commit, REPORT_KEY))
+                .build();
+
+        LOGGER.info("Creating annotations on bitbucket cloud");
+        LOGGER.debug("Create annotations: " + getObjectMapper().writeValueAsString(annotations));
+
+        try (Response response = getClient().newCall(req).execute()) {
+            validate(response);
+        }
+    }
+
+    void deleteReport(String project, String repository, String commit) throws IOException {
+        Request req = new Request.Builder()
+                .delete()
+                .url(format("%s/2.0/repositories/%s/%s/commit/%s/reports/%s", baseUrl(), project, repository, commit, REPORT_KEY))
+                .build();
+
+        LOGGER.info("Deleting existing reports on bitbucket cloud");
+
+        try (Response response = getClient().newCall(req).execute()) {
+            // we dont need to validate the output here since most of the time this call will just return a 404
+        }
+    }
+
+    private OkHttpClient getClient() {
+        return new OkHttpClient.Builder()
+                .addInterceptor(chain -> {
+                    Request newRequest = chain.request().newBuilder()
+                            .addHeader("Authorization", format("Basic %s", getToken()))
+                            .addHeader("Accept", APPLICATION_JSON_MEDIA_TYPE.toString())
+                            .build();
+                    return chain.proceed(newRequest);
+                })
+                .build();
+    }
+
+    void validate(Response response) throws IOException {
+        if (!response.isSuccessful()) {
+            String error = null;
+            if (response.body() != null) {
+                error = response.body().string();
+            }
+            throw new RuntimeException("Bitbucket returned the following error: " + error);
```

Comment:
Any chance of using something slightly less generic than a `RuntimeException`. An `IllegalStateException` may suit if you don't have anything more specific.